### PR TITLE
Add basic tests for the Linux integration

### PR DESCRIPTION
**Pull request by jalvz**
## What does this PR do?

Adds a Linux integration test scenario. This test only checks that the integration is installed, the right data stream is created, and some data is ingested.

Let me know if this should cover something else.

NOTE: I removed the Git pre-commit hook. `go lint ./...`, `make golint`, and `golangci-lint run` all pass successfully, but the golint step from the pre-commit hook keeps failing. I don't know what else I need to do to lint this, so hoping CI or someone can tell me. 

## Why is it important?

To test the Linux integration!

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have run the Unit tests for the CLI, and they are passing locally
- [X] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
~~- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)~~

## How to test this PR locally

`godog -t @linux_integration`

## Related issues

Implements #825

### CODE REVIEW

File: e2e/_suites/fleet/fleet.go [hunk 1]
```diff
@@ -1127,6 +1127,81 @@ func (fts *FleetTestSuite) upgradeAgent(version string) error {
 	return nil
 }
 
+func (tfs *FleetTestSuite) checkDataStream() error {
```
Comment (mdelapenya) [inline e2e/_suites/fleet/fleet.go:1130]: This method is great! I can imagine up-coming tests for integrations in which this method would be abstracted so that it supports passing `exists` field names and `term` terms.

File: e2e/_suites/fleet/integrations.go [hunk 2]
```diff
@@ -17,18 +18,74 @@ const elasticEnpointIntegrationTitle = "Endpoint Security"
...
+
 // addIntegrationToPolicy sends a POST request to Fleet adding an integration to a configuration
 func addIntegrationToPolicy(integrationPackage IntegrationPackage, policyID string) (string, error) {
-	name := integrationPackage.name + "-test-name"
-	description := integrationPackage.title + "-test-description"
 
-	body, err := kibanaClient.AddIntegrationToPolicy(integrationPackage.name, name, integrationPackage.title, description, integrationPackage.version, policyID)
+	policy := Policy{
+		AgentPolicyID: policyID,
+		Name:          integrationPackage.Name + "-test-" + "-" + policyID,
```
Comment (mdelapenya) [inline e2e/_suites/fleet/integrations.go:56]: Consistency with previous behaviour:
```suggestion
		Name:          integrationPackage.Name + "-test-name-" + policyID,
```

File: e2e/_suites/fleet/integrations.go [hunk 3]
```diff
@@ -17,18 +18,74 @@ const elasticEnpointIntegrationTitle = "Endpoint Security"
...
+}
+
+type PolicyInput struct {
+	Type    string         `json:"type"`
+	Enabled bool           `json:"enabled"`
+	Streams []interface{}  `json:"streams"`
+	Vars    map[string]Var `json:"vars,omitempty"`
+}
+
+type Var struct {
```
Comment (mdelapenya) [inline e2e/_suites/fleet/integrations.go:46]: The error in golint can be found here: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2FPR-924/detail/PR-924/1/tests
>e2e/_suites/fleet/integrations.go:27:6: exported type Policy should have comment or be unexported
e2e/_suites/fleet/integrations.go:39:6: exported type PolicyInput should have comment or be unexported
e2e/_suites/fleet/integrations.go:46:6: exported type Var should have comment or be unexported

```suggestion
// Policy represents an integration policy
type Policy struct {
	ID            string             `json:"id,omitempty"`
	Name          string             `json:"name"`
	Description   string             `json:"description"`
	Namespace     string             `json:"namespace"`
	Enabled       bool               `json:"enabled"`
	AgentPolicyID string             `json:"policy_id"`
	OutputID      string             `json:"output_id"`
	Inputs        []PolicyInput      `json:"inputs"`
	Package       IntegrationPackage `json:"package"`
}

// PolicyInput represents an input for the integration policy
type PolicyInput struct {
	Type    string         `json:"type"`
	Enabled bool           `json:"enabled"`
	Streams []interface{}  `json:"streams"`
	Vars    map[string]Var `json:"vars,omitempty"`
}

// Var represents variables in a policy input
type Var struct {
```

File: e2e/_suites/fleet/fleet.go (hunk 1 above)
Comment (mdelapenya) [inline e2e/_suites/fleet/fleet.go:1130]: Consistency in the receiver name

```suggestion
func (fts *FleetTestSuite) checkDataStream() error {
```

File: cli/services/kibana.go [hunk 4]
```diff
@@ -59,22 +61,12 @@ func (k *KibanaClient) withURL(path string) *KibanaClient {
...
-		"output_id":"",
-		"inputs":[],
-		"package":{
-			"name":"` + packageName + `",
-			"title":"` + title + `",
-			"version":"` + version + `"
-		}
-	}`
-
+func (k *KibanaClient) AddIntegrationToPolicy(policy interface{}) (string, error) {
```
Comment (mdelapenya) [inline cli/services/kibana.go:64]: 👍

**Review by mdelapenya — COMMENTED** [review]
LGTM, although it broke both the endpoint integration and the background-processes tests. You can take a look at the logs for the primer here: https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/e2e-tests/pipelines/e2e-testing-mbp/pipelines/PR-924/runs/1/nodes/457/steps/1915/log/?start=0

>[2021-03-23T19:01:03.590Z] time="2021-03-23T19:01:03Z" level=error msg="Could not add integration to configuration" body="{\"statusCode\":400,\"error\":\"Bad Request\",\"message\":\"[request body.inputs]: expected value of type [array] but got [null]\"}" error="POST request failed with 400" payload="{\"name\":\"endpoint-test--fcbc3840-8c09-11eb-8341-9f851332ea20\",\"description\":\"Endpoint Security-test-description\",\"namespace\":\"default\",\"enabled\":true,\"policy_id\":\"fcbc3840-8c09-11eb-8341-9f851332ea20\",\"output_id\":\"\",\"inputs\":null,\"package\":{\"name\":\"endpoint\",\"title\":\"Endpoint Security\",\"version\":\"0.18.0\"}}\n" policy="{ endpoint-test--fcbc3840-8c09-11eb-8341-9f851332ea20 Endpoint Security-test-description default true fcbc3840-8c09-11eb-8341-9f851332ea20  [] { endpoint Endpoint Security 0.18.0 <nil>}}" url="http://localhost:5601/api/fleet/package_policies"

File: e2e/_suites/fleet/integrations.go [hunk 5]
```diff
@@ -17,18 +18,74 @@ const elasticEnpointIntegrationTitle = "Endpoint Security"
...
-	description := integrationPackage.title + "-test-description"
 
-	body, err := kibanaClient.AddIntegrationToPolicy(integrationPackage.name, name, integrationPackage.title, description, integrationPackage.version, policyID)
+	policy := Policy{
+		AgentPolicyID: policyID,
+		Name:          integrationPackage.Name + "-test-" + "-" + policyID,
+		Description:   integrationPackage.Title + "-test-description",
+		Namespace:     "default",
+		Enabled:       true,
+		Package:       integrationPackage,
```
Comment (mdelapenya) [inline e2e/_suites/fleet/integrations.go:60]: This will fix the `null` error while posting the endpoint integration. If you take a look at https://github.com/elastic/e2e-testing/pull/924/files#diff-738d8d068f7c4134a147cd0f4b65a3cc7045391b344ee47e9a17722705b76749L70, you'll find that the previous value was an empty array.
```suggestion
		Package:       integrationPackage,
		Inputs:        []PolicyInput{},
```

**Comment by mdelapenya** [comment]
Great job with the PR @jalvz! I think this code will support adding tests for integrations in a super easy manner, maybe (and simply) abstracting the `checkDataStream` method to dynamically build the query for the integration and the index to be used in searches.

I added a few comments, please let me know wdyt.

**Review by mdelapenya — CHANGES_REQUESTED** [review]
I added a few inline comments. Once they are addressed, this is good to go. thanks for your first wonderful contribution!

File: e2e/_suites/fleet/integrations.go (hunk 5 above)
Comment (jalvz) [reply e2e/_suites/fleet/integrations.go:63]: Good catch, thanks. I had already most of suggestions implemented, which I managed to leave out thanks to my clever misuse of `git add` 🤦 - but this one I missed.

File: e2e/_suites/fleet/fleet.go (hunk 1 above)
Comment (jalvz) [reply e2e/_suites/fleet/fleet.go]: yes!

**Review by mdelapenya — APPROVED** [review]
LGTM!

**Comment by mdelapenya** [comment]
Please use mergify (see https://github.com/elastic/e2e-testing/blob/master/.mergify.yml) for the backports

File: e2e/_suites/fleet/fleet.go [hunk 6]
```diff
@@ -1127,6 +1127,80 @@ func (fts *FleetTestSuite) upgradeAgent(version string) error {
 	return nil
 }
 
+func (fts *FleetTestSuite) checkDataStream() error {
```
Comment (EricDavisX) [inline e2e/_suites/fleet/fleet.go:1130]: do we think it is worthwhile to abstract this a bit now, or wait for later?  I am hopeful we'll have more datastream tests, so at a minimum we might rename it to be specific to Linux if not any more abstracted.

If we do abstract it, it gets a little messier now perhaps, but the re-use could speed up other test additions, if we think the query is generic enough applied.

**Review by EricDavisX — APPROVED** [review]
looks great - we can push this in and build off of it more - thanks so much.

do we need to document any better what explicit validations we have apart from the code itself?  do we think that is read-able (and quick) enough for future automation folks?  It is an open question, not saying it is or isn't.

**Comment by jalvz** [comment]
Thanks @EricDavisX 

> do we need to document any better what explicit validations we have apart from the code itself

Personally I think the `ingest` feature tag is meaningful enough, but i'm happy to follow any existing conventions. I'll get this merged in the meanwhile and push a doc update if needed.

**Comment by EricDavisX** [comment]
And I'll say too, that the 'is there data new within the last minute' is the type of validation we need to insert in many places in the testing here, with the System Policy fields.  This is a great piece of work!

**Comment by jalvz** [comment]
@mdelapenya this PR cant be merged because because of failing CI. I don't see how failing CI is related to this PR. How can I merge this?

**Comment by mdelapenya** [comment]
> @mdelapenya this PR can't be merged because because of failing CI. I don't see how failing CI is related to this PR. How can I merge this?

Oh, let me do it for you. I'll add you to the contributors team so that you can do it by yourself. Sorry about that!

**Comment by mdelapenya** [comment]
Merging, as the test error could be considered flaky, and it's appearing in master
